src/renderer/components/Meta.tsx

```diff
@@ -57,26 +54,75 @@ console.log("Saving to", savePath);
 
 try {
   const data = JSON.parse(readFileSync(savePath, 'utf-8'));
-  initialState = {
-    version: data.version,
-    autoEdit: data.autoEdit,
-    isSelect: data.isSelect,
-    config: data.config,
-    scenes: data.scenes.map((s: any) => new Scene(s)),
-    library: data.library.map((s: any) => new LibrarySource(s)),
-    tags: data.tags.map((t: any) => new Tag(t)),
-    route: data.route.map((s: any) => new Route(s)),
-  };
-
-  // validate initialState is rehydrated from a 'compatible' version
-  if (initialState.version != VERSION)
-  {
-    // ToDo (in future) deal with upgrading incompatible versions 
-    //  For now the version is flattened to the current version.
-    initialState.version = VERSION;
+
+  switch (data.version) {
+
+    // When no version number found in data.json - assume v1.2.0 format
+    // This should fail safe and self heal.
+    case undefined:
+
+      // Preserve the existing file - so as not to destroy user's data
+      archiveFile(savePath);
+
+      // Version:
+      // Explicitly 'upgrading' so set the new version number
+      initialState.version = __VERSION__;
+
+      // Library:
+      // Create Library from aggregate of previous scenes' directories
+      // technique is by text manipulation hack 
+      let libraryString: string = "";
+      let sourceId: number = 0;
+      let directories: string[];
+
+      // Get a unique array of directories from old scenes
+      data.scenes.forEach((oldScene: any) => {
+        directories = [].concat(directories, oldScene.directories).filter((val, ind, arr) => arr.indexOf(val) === ind);
```

Comment:
Each time you're looping here, you're setting directories to the results of this scene. I think you meant to concat to `directories` and not `[]`. The variable `directories` is typed above, but not defined (Could be `const directories = Array<string>();`.
For the record, primitive types are implied, so you don't have to explicitly type `string` or `number`.  🙂